### PR TITLE
WIP: custom row-metadata (#260)

### PR DESCRIPTION
**Pull request by nbenn**
Ad #260

### CODE REVIEW

**Comment by nbenn** [comment]
Happy to add some tests as well, something along the lines of the examples I supplied?

File: R/ctl_colonnade.R [hunk 1]
```diff
@@ -12,13 +12,13 @@ ctl_colonnade <- function(x, has_row_id = TRUE, width = NULL, controller = new_t
...
   # Reserve space for rowid column in each tier
-  if (!is_false(has_row_id)) {
-    rowid <- rif_shaft(n)
-    rowid_width <- get_cell_widths(rowid)
-  } else {
-    rowid <- NULL
+  if (is.null(rowid)) {
     rowid_width <- 0L
+  } else {
+    rowid_width <- get_cell_widths(rowid)
```
Comment (krlmlr) [inline R/ctl_colonnade.R:21]: Should `get_cell_widths(NULL)` return `0L`, to simplify this code?

File: R/ctl_new_pillar.R [hunk 2]
```diff
@@ -110,6 +153,26 @@ ctl_new_compound_pillar <- function(controller, x, width, ..., title = NULL) {
...
+ctl_new_rowid_pillar <- function(controller, x, width, has_row_id, ..., title = NULL) {
+  "!!!!DEBUG ctl_new_rowid_pillar(`v(width)`, `v(title)`)"
+
+  check_dots_empty()
+
+  if (length(width) == 0) {
+    return(NULL)
+  }
+
+  if (is_false(has_row_id)) {
```
Comment (krlmlr) [inline R/ctl_new_pillar.R:169]: I wonder if this should be picked up by the methods instead?

**Review by krlmlr — COMMENTED** [review]
Thanks, the code looks good. Could you please add a snapshot test?

**Comment by krlmlr** [comment]
This has conflicts now, unfortunately.

**Pull request closed without merging**